torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -201,140 +201,231 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
     return size_;
   }
 
-  virtual const std::string getBackendName() const {
-    return "undefined";
-  }
+  // Subclasses must override this method to return the backend name
+  virtual const std::string getBackendName() const = 0;
```

Comment:
Probably it is fairly an edge case since I assume we own all `ProcessGroup`s out there, but technically this is a BC-breaking change. A subclass that did not override `getBackendName()` will now fail to compile due to it being an abstract function.

Reply:
Right. ProcessGroup C++ is currently a private API. I am trying bring it to prototype in the v1.11 release. Will make a few BC-breaking changes during this process.